modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/search/spi/model/query/contributor/DLFileEntryKeywordQueryContributor.java

```diff
@@ -77,14 +81,24 @@ public void contribute(
 					keywords, StringPool.QUOTE);
 
 				if (Validator.isNotNull(exactMatch)) {
-					String notExactKeyword = keywords.replaceFirst(
-						StringPool.QUOTE + exactMatch + StringPool.QUOTE, "");
-
 					fileNameBooleanQuery.add(
 						_getMatchQuery(
 							"fileName", exactMatch, MatchQuery.Type.PHRASE),
 						BooleanClauseOccur.MUST);
 
+					String notExactKeyword = StringPool.BLANK;
+
+					try {
+						notExactKeyword = keywords.replaceFirst(
+							StringPool.QUOTE + exactMatch + StringPool.QUOTE,
```

Comment:
To avoid syntax error in the regex, you should use `Pattern.quote()` to escape any rx especial characters in `exactMatch`:
```
notExactKeyword = keywords.replaceFirst(Pattern.quote(StringPool.QUOTE + ...));
```